cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java

```diff
@@ -380,13 +381,108 @@ private void convertACLineSegmentsToLines(Context context, Set<String> delayedBo
             if (c.valid()) {
                 String node = c.boundaryNode();
                 if (node != null && !context.config().convertBoundary()) {
-                    context.boundary().addAcLineSegmentAtNode(line, node);
-                    delayedBoundaryNodes.add(node);
+                    String lineContainerId = line.getId("Line");
+                    if (lineContainerId != null) {
+                        delayedLines.computeIfAbsent(lineContainerId, l -> new PropertyBags()).add(line);
```

Comment:
Here we are recording ONLY `ACLineSegments` that are contained inside a `cim:Line` AND have one end at the boundary. We could have `ACLineSegments` inside the network, none of the two ends placed at the boundary.